### PR TITLE
Feat/user profile

### PR DESCRIPTION
**Pull request by Arseniia-Damaksina**
## Describe your changes

- I added the sidebar for admin panel;
- Also styled the topbar for admin panel.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-67)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
admin-panel
admin-edit

### CODE REVIEW

File: src/pages/UserProfileAdminPage/SideBar.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+import { StyledListItemButton, StyledSubListItemButton } from './styles';
+
+const listUsers: number = 0;
+const listProducts: number = 1;
+const listChats: number = 2;
+
+const subListBuyers: number = 1;
+const subListVendors: number = 2;
+const subListRequests: number = 3;
+const subListProducts: number = 4;
```
Comment (lyudad) [inline src/pages/UserProfileAdminPage/SideBar.tsx]: better use enum or at least json

File: src/pages/UserProfileAdminPage/styles.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+const backgroundOpacity: number = 0.5;
+
+const hexToRgba = (hex: string, opacity: number) => {
+  /* eslint-disable no-magic-numbers */
+  const r = parseInt(hex.slice(1, 3), hexToDecimal);
+  const g = parseInt(hex.slice(3, 5), hexToDecimal);
+  const b = parseInt(hex.slice(5, 7), hexToDecimal);
+
+  return `rgba(${r}, ${g}, ${b}, ${opacity})`;
+};
```
Comment (lyudad) [inline src/pages/UserProfileAdminPage/styles.ts]: its better make such function as common, as it could be used in on other places also

Comment (Arseniia-Damaksina) [reply src/pages/UserProfileAdminPage/styles.ts]: moved the function to commons

File: src/pages/UserProfileAdminPage/SideBar.tsx (hunk 1 above)
Comment (Arseniia-Damaksina) [reply src/pages/UserProfileAdminPage/SideBar.tsx]: fixed, used object instead, because with enums eslint still complains about magic numbers

**Review by zweroboy1 — APPROVED** [review]

**Pull request merged**